gmplot/gmplot.py

```diff
@@ -274,19 +282,45 @@ def write_map(self,  f):
             '\t\tvar map = new google.maps.Map(document.getElementById("map_canvas"), myOptions);\n')
         f.write('\n')
 
-    def write_point(self, f, lat, lon, color, title):
+    def write_point(self, f, lat, lon, color, title, label):
         f.write('\t\tvar latlng = new google.maps.LatLng(%f, %f);\n' %
                 (lat, lon))
         f.write('\t\tvar img = new google.maps.MarkerImage(\'%s\');\n' %
                 (self.coloricon % color))
         f.write('\t\tvar marker = new google.maps.Marker({\n')
+        f.write('\t\tlabel: \'%s\',\n' % label)
         f.write('\t\ttitle: "%s",\n' % title)
         f.write('\t\ticon: img,\n')
         f.write('\t\tposition: latlng\n')
         f.write('\t\t});\n')
         f.write('\t\tmarker.setMap(map);\n')
         f.write('\n')
 
+    def write_direction(self, f, orig, dest, waypoints=[]):
+        f.write('\n')
+        f.write('\t\tvar start = new google.maps.LatLng(%f, %f);\n' % (float(orig.split(",")[0]), float(orig.split(",")[1])))
+        f.write('\t\tvar end = new google.maps.LatLng(%f, %f);\n' % (float(dest.split(",")[0]), float(dest.split(",")[1])))
+        f.write('\t\tvar mid = [];\n')
+
+        for waypoint in waypoints:
+		    f.write('\t\tmid.push({ location: "%f,%f", stopover: true });\n' % (waypoint[0], waypoint[1]))
```

Comment:
Instead of having the JavaScript code build `mid`, let's build it in this function so that the resultant HTML portion looks like:
```
waypoints: [
    {location: "<lat,lng>", stopover: true},
    {location: "<lat,lng>", stopover: true},
    {location: "<lat,lng>", stopover: true},
]
```